sql/parser/builtins.go

```diff
@@ -66,9 +67,16 @@ func (b builtin) match(types typeList) bool {
 // sorted please.
 var builtins = map[string][]builtin{
 	// TODO(XisiHuang): support encoding, i.e., length(str, encoding).
-	"length": {stringBuiltin1(func(s string) (Datum, error) {
-		return DInt(len(s)), nil
-	}, DummyInt)},
+	"length": {
+		stringBuiltin1(func(s string) (Datum, error) {
+			return DInt(utf8.RuneCountInString(s)), nil
+		}, DummyInt),
+		bytesBuiltin1(func(s string) (Datum, error) {
+			return DInt(len(s)), nil
+		}, DummyInt),
+	},
+
+	// TODO(pmattis): What string functions should also support bytesType?
```

Comment:
And which ones should *only* be bytesType? If we want to be strict about it, functions like sha256() and to_hex() should only work on bytes.